unicore/hub/common/validators.py

```diff
@@ -0,0 +1,22 @@
+import colander
+
+
+def username_validator(node, value):
+    # http://stackoverflow.com/questions/1597743/what-restrictions-should-i-impose-on-usernames
+    pass  # TODO - must raise colander.Invalid
+
+
+def app_data_validator(node, value):
+    if not isinstance(value, dict):
+        raise colander.Invalid(node,
+                               '%r is not a dictionary' % (value, ))
+
+    for key in value.keys():
+        if not isinstance(key, int) or key < 0:
```

Comment:
any reason why we're restricting `app_id` to ints? I'm thinking alphanumerics would be easier to work with.